tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_included_record_parameters.bal

```diff
@@ -277,6 +283,12 @@ function testFunctionOfFunctionTypedParamWithIncludedRecordParam20() {
     assertEquality(5, val);
 }
 
+function testFunctionOfFunctionTypedParamWithIncludedRecordParam21() {
+    int[] x = [1, 2];
+    int[] val = functionOfFunctionTypedParamWithIncludedRecordParam21(a = [1, 2]);
```

Comment:
Shall we also add a test where the contextually expected type is used? e.g., `[1, 2]` where a `float[]` is expected.

Reply:
Added.